spec/integration/transport/ssh_spec.rb

```diff
@@ -337,6 +337,23 @@ def make_target(host_: hostname, port_: port, conf: config)
     end
   end
 
+  context "with non-sudo executable", sudo: true do
+    let(:config) {
+      mk_config('host-key-check' => false, 'sudo-executable' => 'fake',
+                'run-as' => user, user: bash_user, password: bash_password)
+    }
+    let(:target) { make_target }
+
+    it 'uses the correct executable' do
+      allow_any_instance_of(Net::SSH::Connection::Channel).to receive(:wait).and_return('')
+      expect_any_instance_of(Net::SSH::Connection::Channel).to receive(:exec)
+        .with("fake -S -u bolt -p \\[sudo\\]\\ Bolt\\ needs\\ to\\ run\\ as\\ another\\ "\
```

Comment:
Since we have this test I think verifying that the executable gets escaped would be good. `sudo -n` seems like a good thing to test this with, and basically verify that it errors with `  STDERR:
    sh: 1: sudo\ -n: not found
` instead of `sudo: a password is required`